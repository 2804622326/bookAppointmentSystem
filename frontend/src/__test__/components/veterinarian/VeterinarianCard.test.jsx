import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import VeterinarianCard from '../../../components/veterinarian/VeterinarianCard.jsx';

const vet = { id: 1, firstName: 'A', lastName: 'B', specialization: 'Surg' };

test('renders vet card snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <VeterinarianCard vet={{ ...vet, averageRating: 0, totalReviewers: 0, photo: '' }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
