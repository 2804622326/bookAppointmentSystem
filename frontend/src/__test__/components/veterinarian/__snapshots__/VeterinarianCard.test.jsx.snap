// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders vet card snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-4 xs={12} col"
  >
    <div
      class="accordion"
    >
      <div
        class="accordion-item"
      >
        <h2
          class="accordion-header"
        >
          <button
            aria-expanded="false"
            class="accordion-button collapsed"
            type="button"
          >
            <div
              class="d-flex align-items-center"
            >
              <a
                href="/"
              >
                <img
                  alt="User photo"
                  class="card-img user-image"
                  src="test-file-stub"
                />
              </a>
            </div>
            <div
              class="flex-grow-1 ml-3 px-5"
            >
              <div
                class="title card-title h5"
              >
                Dr. A B
              </div>
              <div
                class="card-title h5"
              >
                <h6>
                  Surg
                </h6>
              </div>
              <p
                class="review rating-stars card-text"
              >
                Reviews: 
                <span
                  class="me-2 ms-2"
                >
                  <svg
                    color="#ffc107"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(255, 193, 7);"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                    />
                  </svg>
                  <svg
                    color="#ffc107"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(255, 193, 7);"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                    />
                  </svg>
                  <svg
                    color="#ffc107"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(255, 193, 7);"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                    />
                  </svg>
                  <svg
                    color="#ffc107"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(255, 193, 7);"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                    />
                  </svg>
                  <svg
                    color="#ffc107"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(255, 193, 7);"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M528.1 171.5L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6zM388.6 312.3l23.7 138.4L288 385.4l-124.3 65.3 23.7-138.4-100.6-98 139-20.2 62.2-126 62.2 126 139 20.2-100.6 98z"
                    />
                  </svg>
                </span>
                 (0)
              </p>
              <a
                class="link"
                href="/book-appointment/1/new-appointment"
              >
                Book appointment
              </a>
            </div>
          </button>
        </h2>
        <div
          class="accordion-collapse collapse"
        >
          <div
            class="accordion-body"
          >
            <div>
              <a
                class="link-2"
                href="/veterinarian/1/veterinarian"
              >
                See what peopple are saying about
              </a>
               
              <span
                class="margin-left-space ms-2"
              >
                Dr.A
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
